### PR TITLE
Add --public-network-access for `az containerapp env create/update`

### PR DESCRIPTION
**Pull request by Juliehzl**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by yonzhan** [comment]
Please fix CI issues

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 

The CI failed due to following error, but this error is not related to containerapp, is this a known issue?
Thanks!
```

/mnt/vss/_work/1/s/env/bin/az:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  __import__('pkg_resources').require('azure-cli==2.62.0')
Traceback (most recent call last):
  File "/mnt/vss/_work/1/s/env/bin/az", line 4, in <module>
    __import__('pkg_resources').require('azure-cli==2.62.0')
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3282, in <module>
    @_call_aside
     ^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3266, in _call_aside
    f(*args, **kwargs)
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 3295, in _initialize_master_working_set
    working_set = _declare_state('object', 'working_set', WorkingSet._build_master())
                                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 589, in _build_master
    ws.require(__requires__)
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 926, in require
    needed = self.resolve(parse_requirements(requirements))
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 787, in resolve
    dist = self._resolve_dist(
           ^^^^^^^^^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 816, in _resolve_dist
    env = Environment(self.entries)
          ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1014, in __init__
    self.scan(search_path)
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 1046, in scan
    for dist in find_distributions(item):
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2091, in find_on_path
    yield from factory(fullpath)
               ^^^^^^^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2183, in resolve_egg_link
    return next(dist_groups, ())
           ^^^^^^^^^^^^^^^^^^^^^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2179, in <genexpr>
    resolved_paths = (
                     ^
  File "/mnt/vss/_work/1/s/env/lib/python3.11/site-packages/pkg_resources/__init__.py", line 2167, in non_empty_lines
    for line in _read_utf8_with_fallback(path).splitlines():
                ^^^^^^^^^^^^^^^^^^^^^^^^
NameError: name '_read_utf8_with_fallback' is not defined
```

File: src/containerapp/azext_containerapp/containerapp_env_decorator.py [hunk 1]
```diff
@@ -269,3 +283,6 @@ def get_argument_certificate_key_vault_url(self):
 
     def get_argument_p2p_encryption_enabled(self):
         return self.get_param("p2p_encryption_enabled")
+
+    def get_argument_public_network_access(self):
+        return self.get_param("public_network_access")
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/containerapp_env_decorator.py:288]: Blank line at the end for style check.

File: src/containerapp/azext_containerapp/_params.py [hunk 2]
```diff
@@ -161,6 +161,10 @@ def load_arguments(self, _):
         c.argument('environment_name', options_list=['--environment'], help="The environment name.")
         c.argument('resource_group_name', arg_type=resource_group_name_type, id_part=None)
 
+    with self.argument_context('containerapp env') as c:
+        c.argument('public_network_access', arg_type=get_enum_type(['Enabled', 'Disabled']),
+               help="Allow or block all public traffic", is_preview=True)
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_params.py:166]: ERROR: /mnt/vss/_work/1/s/src/containerapp/azext_containerapp/_params.py:166:16: E128 continuation line under-indented for visual indent

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**